src/core/border_router/routing_manager.cpp

```diff
@@ -1158,6 +1160,9 @@ bool RoutingManager::UpdateDiscoveredPrefixes(const RouterAdv::RouteInfoOption &
         ExitNow();
     }
 
+    // Ignore own OMR prefix.
+    VerifyOrExit(mLocalOmrPrefix != prefix);
+
```

Comment:
> If we are not advertising our own OMR prefix (it's not in the mAdvertisedOmrPrefixes list or Network Data), why should we ignore this prefix?

I feel it's always right to ignore local OMR Prefix.

> If a BR in another Network network is using the same OMR prefix, we should add external route for the prefix and it will work, right?

Everything breaks if two Thread networks are using same OMR prefixes. Since the chance is extremely low, should we simply not consider such case?

Reply:
One possible case I can see is that another BR of the same Thread network will be advertising the OMR prefix.

Meanwhile, the owner BR of the OMR prefix may not have synchronized the Network Data (e.g. just rebooted). In such case the owner BR would add OMR prefix to discovered prefixes, which is incorrect.

So, we should always ignore `mLocalOmrPrefix` for safety. Same argument may also applies for `mLocalOnLinkPrefix`.

Reply:
further thoughts? @wgtdkp

Reply:
Fine to me, not a big deal :)